test/Dialect/HW/hw-inliner-options.mlir

```diff
@@ -0,0 +1,66 @@
+// RUN: circt-opt %s --pass-pipeline='builtin.module(hw-flatten-modules{inline-all=false inline-single-use=false inline-small=false inline-empty=false inline-no-outputs=false})' | FileCheck %s --check-prefix=NONE
+// RUN: circt-opt %s --pass-pipeline='builtin.module(hw-flatten-modules{inline-all=false inline-single-use=true inline-small=false inline-empty=false inline-no-outputs=false})' | FileCheck %s --check-prefix=SINGLE
+// RUN: circt-opt %s --pass-pipeline='builtin.module(hw-flatten-modules{inline-all=false inline-single-use=false inline-small=true inline-empty=false inline-no-outputs=false})' | FileCheck %s --check-prefix=SMALL
+// RUN: circt-opt %s --pass-pipeline='builtin.module(hw-flatten-modules{inline-all=false small-threshold=3 inline-single-use=false})' | FileCheck %s --check-prefix=THRESHOLD
+// RUN: circt-opt %s --pass-pipeline='builtin.module(hw-flatten-modules{inline-with-state=true})' | FileCheck %s --check-prefix=STATE
+
+// Test that all inlining heuristics can be controlled via command line options
+
+// NONE-LABEL: hw.module @TestSingleUse
+// SINGLE-LABEL: hw.module @TestSingleUse
+hw.module @TestSingleUse(in %x: i4, out y: i4) {
+  // NONE-NEXT: hw.instance "small" @SmallModule
+  // SINGLE-NEXT: %[[V0:.+]] = comb.add %x, %x
+  // SINGLE-NEXT: hw.output %[[V0]]
+  %0 = hw.instance "small" @SmallModule(a: %x: i4) -> (b: i4)
+  hw.output %0 : i4
+}
+
+hw.module private @SmallModule(in %a: i4, out b: i4) {
+  %0 = comb.add %a, %a : i4
+  hw.output %0 : i4
+}
+
+// SMALL-LABEL: hw.module @TestSmall
+hw.module @TestSmall(in %x: i4, out y: i4) {
+  // SMALL-NEXT: %[[V0:.+]] = comb.add %x, %x
+  // SMALL-NEXT: hw.output %[[V0]]
+  %0 = hw.instance "tiny" @TinyModule(a: %x: i4) -> (b: i4)
+  hw.output %0 : i4
+}
+
+hw.module private @TinyModule(in %a: i4, out b: i4) {
+  %0 = comb.add %a, %a : i4
+  hw.output %0 : i4
+}
```

Comment:
Nit: This is just `@SmallModule`. Though, going off of the comment above, I think you want small to be "large" and tiny to be "small".